presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -26,6 +26,10 @@ standaloneExpression
     : expression EOF
     ;
 
+standalonePathSpecification
+    : pathSpecification EOF
```

Comment:
shouldn't `invokeParser` make sure whole input it consumed?

Reply:
I don't think that works, in general. The presence of the EOF token in the rule affects Antlr's prediction algorithms. Let me give it a try, though.

Reply:
Indeed, it doesn't work. For instance, for a grammar like:

```
a: b
b: X Y?
```

invoking `a` with input "XY" will successfully complete by just parsing X and leaving Y untouched. It appears that's what Antlr will do unless `a` is `b EOF`.

From https://github.com/antlr/antlr4/blob/master/doc/parser-rules.md, 

> Start rules don’t necessarily consume all of the input. They consume only as much input as needed to match an alternative of the rule

> On the other hand, rules that describe entire input files should reference special predefined-token EOF. If they don’t, you might scratch your head for a while wondering why the start rule doesn’t report errors for any input no matter what you give it.